src/aof.c

```diff
@@ -2302,7 +2302,11 @@ int rewriteSlotToAppendOnlyFileRio(rio *aof, int db_num, int hashslot, size_t *k
         if (key_count && ((*key_count)++ & 1023) == 0) {
             long long now = mstime();
             if (now - updated_time >= 1000) {
-                sendChildInfo(CHILD_INFO_TYPE_CURRENT_INFO, *key_count, "AOF rewrite");
+                if (aof->flags & RIO_FLAG_SLOT_MIGRATION_AOF) {
+                    sendChildInfo(CHILD_INFO_TYPE_CURRENT_INFO, *key_count, "Slot migration");
+                } else {
+                    sendChildInfo(CHILD_INFO_TYPE_CURRENT_INFO, *key_count, "AOF rewrite");
+                }
```

Comment:
this changes introduce unnecessary coupling and adds cognitive load on the future developer. I don't think `aof.c` should understand what "slot migration" is. the `pname` should be opaque to aof.c and aof.c should just blindly passes along.
```suggestion
                sendChildInfo(CHILD_INFO_TYPE_CURRENT_INFO, *key_count, aof->pname);
```

Reply:
The rdbSaveDb in rdb.c also use this trick (or similar flags trick). aof.c will anyway need to understand the "slot migration" in the furture as long as we use this format (like module event something like that). We may need new aofflags in that time. 
```
ssize_t rdbSaveDb(rio *rdb, int dbid, int rdbflags, long *key_counter) {
    ssize_t written = 0;
    ssize_t res;
    kvstoreIterator *kvs_it = NULL;
    static long long info_updated_time = 0;
    char *pname = (rdbflags & RDBFLAGS_AOF_PREAMBLE) ? "AOF rewrite" : "RDB";
```

Reply:
Do you want to add the pname to rio? I think we should either use rio->flags (although it is a bit less elegant) or similar to rdbflags we use a new aofflags

Reply:
@enjoy-binbin We talked about this in monday meeting. We can punt this, but we want to merge this PR for rc2.

Reply:
I realize that rewriteSlotToAppendOnlyFileRio is currently only used in the slot migration, so i am gonna remove the flags first. We can change it again in the future if needed. @murphyjacob4 please ack again so that we can merge it.

Reply:
Ack, seems reasonable, IMO anyone reusing the function can add the parameterization then